File: Go/FizzBuzz.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	fmt.Println("FizzBuzz V1:")
	fizzBuzzV1()
	fmt.Println("FizzBuzz V2:")
	fizzBuzzV2()
}

func fizzBuzzV1() {
	for i := 1; i <= 100; i++ {
		if i%15 == 0 {
			fmt.Println("FizzBuzz")
		} else if i%5 == 0 {
			fmt.Println("Buzz")
		} else if i%3 == 0 {
			fmt.Println("Fizz")
		} else {
			fmt.Println(i)
		}
	}
}

func fizzBuzzV2() {
	var out string = ""
	for i := 1; i <= 100; i++ {
		out = ""
		if i%3 == 0 {
			out += "Fizz"
		}
		if i%5 == 0 {
			out += "Buzz"
		}
		if out == "" {
			out = strconv.Itoa(i)
		}
		fmt.Println(out)
	}
}

File: Kotlin/FizzBuzz.kt
fun main() {
    println("FizzBuzzV1:")
    fizzBuzzV1()
    println("FizzBuzzV2:")
    fizzBuzzV2()
}

fun fizzBuzzV1() {
    for (i in 1..100) {
        if (i % 15 == 0) {
            println("FizzBuzz")
        } else if (i % 5 == 0) {
            println("Buzz")
        } else if (i % 3 == 0) {
            println("Fizz")
        } else {
            println(i)
        }
    }
} 

fun fizzBuzzV2() {
    var out = ""
    for (i in 1..100) {
        out = ""
        if (i % 3 == 0) {
            out += "Fizz"
        }
        if (i % 5 == 0) {
            out += "Buzz"
        }
        if (out == "") {
            out = i.toString()
        }
        println(out)
    }
}